app/assets/javascripts/discourse/tests/acceptance/sidebar-user-categories-section-test.js

```diff
@@ -1151,5 +1224,65 @@ acceptance(
         "links to the latest topics list for the category because there are no unread or new topics"
       );
     });
+
+    test("category link href is always the latest topics list when sidebar_link_to_filtered_list is false", async function (assert) {
+      const categories = Site.current().categories;
+      const category1 = categories[0];
+      const category2 = categories[1];
+      const category3 = categories[2];
+
+      updateCurrentUser({
+        sidebar_category_ids: [category1.id, category2.id, category3.id],
+        user_option: {
+          sidebar_link_to_filtered_list: false,
+        },
+      });
+
+      this.container.lookup("service:topic-tracking-state").loadStates([
+        {
+          topic_id: 1,
+          highest_post_number: 1,
+          last_read_post_number: null,
+          created_at: "2022-05-11T03:09:31.959Z",
+          category_id: category1.id,
+          notification_level: null,
+          created_in_new_period: true,
+          treat_as_new_topic_start_date: "2022-05-09T03:17:34.286Z",
+        },
+        {
+          topic_id: 2,
+          highest_post_number: 12,
+          last_read_post_number: 11,
+          created_at: "2020-02-09T09:40:02.672Z",
+          category_id: category2.id,
+          notification_level: 2,
+          created_in_new_period: false,
+          treat_as_new_topic_start_date: "2022-05-09T03:17:34.286Z",
+        },
+      ]);
+
+      await visit("/");
+
+      assert.true(
```

Comment:
```suggestion
      assert.ok(
```

Minor for consistency sake because we use `assert.ok` everywhere else.

Reply:
I think I read somewhere (probably a PR review) that we now prefer `assert.true(...)`/`assert.false(...)` over `assert.ok(...)` and `assert.ok(!...)` because the former gives better failure messages.

Reply:
That's `true` 😄 those assertions are preferred (I promise to update the testing guide soon 😅) The main benefit is stricter type checking. We had (probably still have) some _typos_ where `ok()` was used instead instead of `equal()`: `assert.ok(someString, "expected")` and those pass (even though string was incorrect 😛) while `true()` requires, well, `true`